src/ConnectedMode/Suppressions/SuppressionIssueStoreUpdater.cs

```diff
@@ -30,16 +37,121 @@ internal interface ISuppressionIssueStoreUpdater
         /// <summary>
         /// Fetches all available suppressions from the server and updates the server issues store
         /// </summary>
-        void FetchAllServerSuppressions();
+        Task UpdateAllServerSuppressionsAsync();
 
         /// <summary>
         /// Fetches suppressions from the server from the specified timestamp onwards and updates the issues store
         /// </summary>
-        void FetchServerSuppressions(DateTimeOffset fromTimestamp);
+        Task UpdateServerSuppressionsAsync(DateTimeOffset fromTimestamp);
 
         /// <summary>
         /// Clears all issues from the store
         /// </summary>
         void Clear();
     }
+
+    [Export(typeof(ISuppressionIssueStoreUpdater))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class SuppressionIssueStoreUpdater : ISuppressionIssueStoreUpdater, IDisposable
```

Comment:
This class contains a subset of the functionality that was/is in `SonarQubeIssueProvider` i.e. fetching all suppressions from the server.

I've dropped a couple of things:
* all of the module-related stuff, since I don't think we need it any more
* all of the timer-related code, since that will be handled by a different class